pcsx2/GS/GSState.cpp

```diff
@@ -1656,6 +1671,40 @@ void GSState::FlushPrim()
 			ASSERT((int)unused < GSUtil::GetVertexCount(PRIM->PRIM));
 		}
 
+		// Texel coordinate rounding
+		// Helps Manhunt (lights shining through objects).
+		// Can help with some alignment issues when upscaling too, and is for both Software and Hardware renderers.
+		// Sometimes hardware doesn't get affected, likely due to the difference in how GPU's handle textures (Persona minimap).
+		if (m_env.PRIM.TME)
+		{
+			if (!m_env.PRIM.FST) // STQ's
+			{
+				// ST's have the lowest 9 bits rounding down (from hardware tests)
+				for (u32 i = 0; i < m_vertex.tail; i++)
+				{
+					// Don't worry, I hate me too. If there's a modern way to do bit manipulation on floats, I'd love to hear it!
+					GSVertex* v = &m_vertex.buff[i];
+					u32 S = *(u32*)&v->ST.S;
+					u32 T = *(u32*)&v->ST.T;
```

Comment:
```suggestion
					u32 T = std::bit_cast<u32>(v->ST.T);
```

(and `#include <bit>` at the top)

Reply:
Thank you <3